src/app/chat/views/user_list.nim

```diff
@@ -0,0 +1,118 @@
+import NimQml, Tables, json, chronicles, sequtils
+import ../../../status/status
+import ../../../status/accounts
+import ../../../status/chat
+import ../../../status/chat/[message]
+import strutils
+
+type
+  UserListRoles {.pure.} = enum
+    UserName = UserRole + 1
+    LastSeen = UserRole + 2
+    PublicKey = UserRole + 3
+    Alias = UserRole + 4
+    LocalName = UserRole + 5
+    Identicon = UserRole + 6
+
+  User = object
+    username: string
+    alias: string
+    localName: string
+    lastSeen: string
+    identicon: string
+
+QtObject:
+  type
+    UserListView* = ref object of QAbstractListModel
+      status: Status
+      users: seq[string]
+      userDetails: OrderedTable[string, User]
+
+  proc delete(self: UserListView) =
+    self.userDetails.clear()
+    self.QAbstractListModel.delete
+
+  proc setup(self: UserListView) =
+    self.QAbstractListModel.setup
+
+  proc newUserListView*(status: Status): UserListView =
+    new(result, delete)
+    result.userDetails = initOrderedTable[string, User]()
+    result.users = @[]
+    result.status = status
+    result.setup
+
+  method rowCount*(self: UserListView, index: QModelIndex = nil): int = self.users.len
+
+  method data(self: UserListView, index: QModelIndex, role: int): QVariant =
+    if not index.isValid:
+      return
+    if index.row < 0 or index.row >= self.users.len:
+      return
+    
+    let user = self.users[index.row]
```

Comment:
`user` is a pubkey I think? Can we rename it to `pubkey` or something in that case?